state/accountsDB.go

```diff
@@ -1356,6 +1357,57 @@ func (adb *AccountsDB) Close() error {
 	return adb.storagePruningManager.Close()
 }
 
+// PrintStatsForRootHash will print trie statistics for the given rootHash
+func (adb *AccountsDB) PrintStatsForRootHash(rootHash []byte) {
+	stats := statistics.NewTrieStatisticsCollector()
+	mainTrie := adb.getMainTrie()
+
+	tr, ok := mainTrie.(common.TrieStats)
+	if !ok {
+		log.Error(fmt.Sprintf("invalid trie, type is %T", mainTrie))
+		return
+	}
+
+	collectStats(tr, stats, rootHash, nil)
+
+	leavesChannel := make(chan core.KeyValueHolder, leavesChannelSize)
+	err := mainTrie.GetAllLeavesOnChannel(leavesChannel, context.Background(), rootHash, keyBuilder.NewDisabledKeyBuilder())
+	if err != nil {
+		log.Error(err.Error())
+		return
+	}
+
+	for leaf := range leavesChannel {
+		account := &userAccount{}
+		err = adb.marshaller.Unmarshal(account, leaf.Value())
+		if err != nil {
+			log.Trace("this must be a leaf with code", "err", err)
+			continue
+		}
+
+		if len(account.RootHash) == 0 {
+			continue
+		}
+
+		collectStats(tr, stats, account.RootHash, account.Address)
+	}
+
+	stats.Print()
+}
+
+func collectStats(tr common.TrieStats, stats common.TriesStatisticsCollector, rootHash []byte, address []byte) {
+	trieStats, err := tr.GetTrieStats(rootHash)
+	if err != nil {
+		log.Error(err.Error())
+		return
+	}
+	trieStats.RootHash = rootHash
+	trieStats.Address = address
+	stats.Add(trieStats)
+
+	log.Debug(strings.Join(trieStats.ToString(), " "))
```

Comment:
do we need this debug line here if we do the stats.Print ?

Reply:
stats.Print() prints only the top 10 tries (by size and by depth). This will print all the tries, and the the stats.Print() will print the top 10. This should be used if we need info about all of the tries. Can be refactored if deemed necessary.